core/services/ocrcommon/adapters.go

```diff
@@ -71,3 +76,66 @@ func (c *OCR3ContractTransmitterAdapter) Transmit(ctx context.Context, digest oc
 func (c *OCR3ContractTransmitterAdapter) FromAccount() (ocrtypes.Account, error) {
 	return c.ct.FromAccount()
 }
+
+var _ ocr3types.OnchainKeyring[[]byte] = (*OCR3OnchainKeyringMultiChainAdapter)(nil)
+
+type OCR3OnchainKeyringMultiChainAdapter struct {
+	ks         keystore.OCR2
+	st         ocr2.OCR2OnchainSigningStrategy
+	keyBundles map[string]ocr2key.KeyBundle
+	publicKey  ocrtypes.OnchainPublicKey
+}
+
+func NewOCR3OnchainKeyringMultiChainAdapter(ks keystore.OCR2, st ocr2.OCR2OnchainSigningStrategy) (*OCR3OnchainKeyringMultiChainAdapter, error) {
```

Comment:
@vyzaldysanchez This is pretty good! But I would loosen the coupling a bit here -- the config should provide a mapping of names -> keyBundleIds (which we've validated are valid), and that's it.

We shouldn't depend on eg. types.SupportedRelays since that will break existing jobs if we add a new relay without changing updating the job config.